x509util/certificate_request_test.go

```diff
@@ -298,6 +303,225 @@ func TestCertificateRequest_GetCertificateRequest(t *testing.T) {
 	}
 }
 
+func TestCertificateRequest_GetCertificateRequest_challengePassword(t *testing.T) {
+	rsaPEM, err := os.ReadFile("testdata/rsa.key")
+	require.NoError(t, err)
+
+	block, _ := pem.Decode(rsaPEM)
+	rsaKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	require.NoError(t, err)
+
+	expectedPrintableString, err := os.ReadFile("testdata/challengePassword.csr")
+	require.NoError(t, err)
+	expectedUTF8String, err := os.ReadFile("testdata/challengePasswordUTF8.csr")
+	require.NoError(t, err)
+
+	tests := []struct {
+		name      string
+		cr        *CertificateRequest
+		want      []byte
+		assertion assert.ErrorAssertionFunc
+	}{
+		{"ok", &CertificateRequest{
+			Subject:            Subject{CommonName: "commonName"},
+			DNSNames:           []string{"foo.com"},
+			EmailAddresses:     []string{"root@foo.com"},
+			IPAddresses:        []net.IP{net.ParseIP("::1")},
+			URIs:               []*url.URL{{Scheme: "mailto", Opaque: "root@foo.com"}},
+			SANs:               []SubjectAlternativeName{{Type: "dns", Value: "bar.com"}},
+			ChallengePassword:  "challengePassword",
+			Extensions:         []Extension{{ID: []int{1, 2, 3, 4}, Critical: true, Value: []byte("foobar")}},
+			SignatureAlgorithm: SignatureAlgorithm(x509.SHA256WithRSA),
+			Signer:             rsaKey,
+		}, expectedPrintableString, assert.NoError},
+		{"ok UTF8String", &CertificateRequest{
+			Subject:            Subject{CommonName: "commonName"},
+			DNSNames:           []string{"foo.com"},
+			EmailAddresses:     []string{"root@foo.com"},
+			IPAddresses:        []net.IP{net.ParseIP("::1")},
+			URIs:               []*url.URL{{Scheme: "mailto", Opaque: "root@foo.com"}},
+			SANs:               []SubjectAlternativeName{{Type: "dns", Value: "bar.com"}},
+			ChallengePassword:  "🔐",
+			Extensions:         []Extension{{ID: []int{1, 2, 3, 4}, Critical: true, Value: []byte("foobar")}},
+			SignatureAlgorithm: SignatureAlgorithm(x509.SHA256WithRSA),
+			Signer:             rsaKey,
+		}, expectedUTF8String, assert.NoError},
+		{"fail challengePAssword", &CertificateRequest{
```

Comment:
```suggestion
		{"fail challengePassword", &CertificateRequest{
```

Reply:
Fixed in https://github.com/smallstep/crypto/commit/7a840550c13c1d2fe86d3bd5e6b436f28ff07634